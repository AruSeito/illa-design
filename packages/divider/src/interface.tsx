import { HTMLAttributes } from "react"

export type DividerDirection = "vertical" | "horizontal"
export type DividerVariant = "solid" | "dashed" | "dotted" | "double"

export interface DividerProps extends HTMLAttributes<HTMLDivElement> {
  direction?: DividerDirection
  variant?: DividerVariant
}
