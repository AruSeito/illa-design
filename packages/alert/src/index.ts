export * from "./alert"
export * from "./interface"
