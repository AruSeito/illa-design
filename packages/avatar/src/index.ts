export * from "./avatar"
export * from "./interface"
export * from "./avatar-group"