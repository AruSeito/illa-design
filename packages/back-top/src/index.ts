export * from "./backtop"
export * from "./interface"
export * from "./style"
