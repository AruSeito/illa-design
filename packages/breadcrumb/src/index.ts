export * from "./interface"
export * from "./breadcrumb"
export * from "./breadcrumbItem"
