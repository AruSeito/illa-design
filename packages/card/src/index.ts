export * from "./card"
export * from "./interface"
export * from "./grid"
export * from "./meta"
