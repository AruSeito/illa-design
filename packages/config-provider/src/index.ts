export * from "./config-provider"
export * from "./interface"

// locale
export * from "./locale/en-US"
export * from "./locale/zh-CN"
export * from "./locale/def"
