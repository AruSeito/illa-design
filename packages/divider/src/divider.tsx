/** @jsxImportSource @emotion/react */
import * as React from "react"
import { forwardRef } from "react"
import { css, SerializedStyles } from "@emotion/react"
import { DividerProps, DividerVariant } from "./interface"
import { globalColor, illaPrefix } from "@illa-design/theme"

function applyDividerContainerHorizontal(variant: DividerVariant): SerializedStyles {
  return css`
    width: 100%;
    border-bottom: 1px ${variant} ${globalColor(`--${illaPrefix}-gray-08`)};
  `
}

function applyDividerContainerVertical(variant: DividerVariant): SerializedStyles {
  return css`
    height: 1em;
    min-width: 0;
    max-width: 0;
    border-right: 1px ${variant} ${globalColor(`--${illaPrefix}-gray-08`)};
  `
}

export const Divider = forwardRef<HTMLDivElement, DividerProps>((props, ref) => {

  const {
    direction = "vertical",
    variant = "solid",
    ...otherProps
  } = props

  let dividerCss: SerializedStyles
  switch (direction) {
    case "vertical":
      dividerCss = applyDividerContainerVertical(variant)
      break
    case "horizontal":
      dividerCss = applyDividerContainerHorizontal(variant)
      break
  }

  return <div ref={ref} {...otherProps}>
    <div css={dividerCss} />
  </div>
})
