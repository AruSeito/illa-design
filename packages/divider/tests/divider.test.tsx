import { render } from "@testing-library/react"
import { Divider } from "../src"

test("Divider renders horizontal", () => {
  const { asFragment } = render(<Divider direction="horizontal" variant="solid" />)
  expect(asFragment()).toMatchSnapshot()
})

test("Divider renders vertical", () => {
  const { asFragment } = render(<Divider direction="vertical" variant="dashed" />)
  expect(asFragment()).toMatchSnapshot()
})