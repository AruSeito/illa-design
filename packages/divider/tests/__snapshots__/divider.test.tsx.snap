// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders horizontal 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  border-bottom: 1px solid #ebebeb;
}

<div>
    <div
      class="emotion-0"
    />
  </div>
</DocumentFragment>
`;
